dot/core/messages.go

```diff
@@ -63,6 +64,8 @@ func (s *Service) HandleTransactionMessage(peerID peer.ID, msg *network.Transact
 						Reason: peerset.BadTransactionReason,
 					}, peerID)
 				}
+
+				hasInvalidTxn = true
```

Comment:
Ok this is out of scope, but it would be really nice if you could move this mega confusing `func() error` outside of this function? And you can pass things as arguments instead.

Reply:
I fully support this, it was a point of confusion for me while testing this lol

Reply:
I bet it was 😄